packages/toolpad-app/src/toolpad/Home.tsx

```diff
@@ -271,51 +368,131 @@ function AppCard({ app, activeDeployment, onDelete }: AppCardProps) {
-      <Menu
-        id="basic-menu"
-        anchorEl={menuAnchorEl}
-        open={menuOpen}
-        onClose={handleMenuClose}
-        MenuListProps={{
-          'aria-labelledby': 'basic-button',
-          dense: true,
-        }}
-      >
-        <MenuItem onClick={handleRenameClick}>
-          <ListItemIcon>
-            <DriveFileRenameOutlineIcon />
-          </ListItemIcon>
-          <ListItemText>Rename</ListItemText>
-        </MenuItem>
-        <MenuItem onClick={handleDeleteClick}>
-          <ListItemIcon>
-            <DeleteIcon />
-          </ListItemIcon>
-          <ListItemText>Delete</ListItemText>
-        </MenuItem>
-      </Menu>
+      <AppMenu
+        menuAnchorEl={menuAnchorEl}
+        menuOpen={menuOpen}
+        handleMenuClose={handleMenuClose}
+        handleRenameClick={handleRenameClick}
+        handleDeleteClick={handleDeleteClick}
+      />
+    </React.Fragment>
+  );
+}
+
+interface AppRowProps {
+  app?: App;
+  activeDeployment?: Deployment;
+  onDelete?: () => void;
+}
+
+function AppRow({ app, activeDeployment, onDelete }: AppRowProps) {
+  const [menuAnchorEl, setMenuAnchorEl] = React.useState<null | HTMLElement>(null);
+
+  const menuOpen = Boolean(menuAnchorEl);
+
+  const [editingName, setEditingName] = React.useState<boolean>(false);
+
+  const handleOptionsClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setMenuAnchorEl(event.currentTarget);
+  };
+
+  const handleMenuClose = React.useCallback(() => {
+    setMenuAnchorEl(null);
+  }, []);
+
+  const handleRenameClick = React.useCallback(() => {
+    setMenuAnchorEl(null);
+    setEditingName(true);
+  }, []);
+
+  const handleDeleteClick = React.useCallback(() => {
+    setMenuAnchorEl(null);
+    if (onDelete) {
+      onDelete();
+    }
+  }, [onDelete]);
+
+  return (
+    <React.Fragment>
+      <TableRow hover role="row">
+        <TableCell component="th" scope="row">
+          <AppNameEditable
+            loading={Boolean(!app)}
+            app={app}
+            editing={editingName}
+            setEditing={setEditingName}
+            description={app ? `Edited ${getReadableDuration(app.editedAt)}` : <Skeleton />}
+          />
+        </TableCell>
+        <TableCell align="right">
+          <Stack direction="row" spacing={1} justifyContent={'flex-end'}>
+            <AppEditButton app={app} />
+            <AppOpenButton app={app} activeDeployment={activeDeployment} />
+            <AppOptions menuOpen={menuOpen} onClick={handleOptionsClick} />
+          </Stack>
+        </TableCell>
+      </TableRow>
+      <AppMenu
+        menuAnchorEl={menuAnchorEl}
+        menuOpen={menuOpen}
+        handleMenuClose={handleMenuClose}
+        handleRenameClick={handleRenameClick}
+        handleDeleteClick={handleDeleteClick}
+      />
     </React.Fragment>
   );
 }
+interface TabPanelProps {
+  children?: React.ReactNode;
+  index: number;
+  value: number;
+}
+
+function TabPanel(props: TabPanelProps) {
```

Comment:
In [other places](https://github.com/mui/mui-toolpad/blob/54095761505e65617bb043f1b499c6095d380858/packages/toolpad-app/src/toolpad/AppEditor/BindingEditor.tsx#L245) we use the MUI `TabContext` and `TabPanel`.